ethereum/core/src/main/java/tech/pegasys/teku/core/signatures/record/ValidatorSigningRecord.java

```diff
@@ -52,11 +60,24 @@ public ValidatorSigningRecord(
   }
 
   public static ValidatorSigningRecord fromBytes(final Bytes data) {
-    return ValidatorSigningRecordSerialization.readRecord(data);
+    try {
+      final SlashingProtectionRecord spr =
+          yamlProvider.getObjectMapper().readValue(data.toArray(), SlashingProtectionRecord.class);
+      return ValidatorSigningRecord.fromSlashingProtectionRecord(spr);
+    } catch (IOException e) {
+      throw new UncheckedIOException(e);
+    }
   }
 
   public Bytes toBytes() {
-    return ValidatorSigningRecordSerialization.writeRecord(this);
+    try {
+      return Bytes.wrap(
+          yamlProvider.getObjectMapper().writeValueAsBytes(this.toSlashingProtectionRecord(null)));
+    } catch (JsonGenerationException | JsonMappingException e) {
+      throw new IllegalStateException("Failed to serialize ValidatorSigningRecord", e);
+    } catch (IOException e) {
+      throw new UncheckedIOException(e);
+    }
   }
```

Comment:
Would be good to keep these as a simple delegation so that this class is focussed on being the data model and the details of serialisation are handled separately.

But they've also wound up only being used by tests which suggests the tests aren't actually testing what the production code now does.

Reply:
My original implementation had this all removed pretty much, and the tests actually wrote to file, and checked the file contents. It as closer to production because it was following it more faithfully, but we didn't want to remove the sync data writer, so i went back to this, but i think i preferred the old one more...